connexion/spec.py

```diff
@@ -269,7 +269,7 @@ def base_path(self):
 
     @base_path.setter
     def base_path(self, base_path):
-        base_path = canonical_base_path(base_path)
```

Comment:
A bit wider comment: the `property` itself still uses the `canonical_base_path` function, so the "getter" returns a different value, there's also the added complexity that the base path is used as Flask blueprint name, which has its own rules